server/container/util/src/test/java/org/apache/james/util/ReactorUtilsTest.java

```diff
@@ -44,6 +47,28 @@
 
 class ReactorUtilsTest {
     static final int BUFFER_SIZE = 5;
+    
+    @Nested
+    class Throttling {
+        @Test
+        void throttleShouldApplyMaxSize() {
```

Comment:
We miss a test where processing takes a lot of time (eg more than throttling window). We need to ensure some form of back-pressure to avoid crashing the system...